### PR TITLE
Fixes #531 - Title retrieving is failing with multiple use case

### LINKED ISSUE #531: Page title fetching is still failing in some cases
Hi,
I recently installed Shaarli (v0.6.5) on a private server (running PHP 7.0.5), in order to save a large amount of links from opened browser tabs. As I imported links manually, I noticed that the link’s title did not always appear properly (nothing, only partially or breaking the UI).  That happened quite often (about 90 broken titles on more than 520 links). As these issues seemed to have been fixed in #410 and #512, I tried to reiterate the process with a local instance (same PHP version), using the most recent development version at that time (11609d9). The result was a bit better (no UI breakage), but about 50 links are still broken, in different ways.
These are the following:

No title appears:
- http://www.latitud-argentina.com/
- http://lesacados.com/
- http://www.a-contresens.net/
- http://www.leboncoin.fr/annonces/offres/ile_de_france/occasions/?th=1&q=test
- https://riunet.upv.es/
- http://pcsx2.net/
- https://www.linagora.com/
- http://coderstats.net/
- http://boilingsteam.com/
- https://www.amazon.fr/gp/cart/view.html/ref=nav_cart
- http://www.lesnumeriques.com/carte-memoire/comparatif-cartes-micro-sd-sd-compact-flash-a340.html#filters/283c=0&283=433&pricex=10&pricey=383&pricen=0
- https://developer.apple.com/library/mac/documentation/UserExperience/Reference/ScreenSaver/ObjC_classic/index.html
- https://fr.atlassian.com/git/tutorials/comparing-workflows/forking-workflow
- https://www.facebook.com/
- http://ast2015.acroe-ica.org/
- http://android.izzysoft.de/
- https://static.societegenerale.fr/
- http://societegenerale.fr/
- http://www.ldlc.com/
- https://www.hiawatha-webserver.org/
- https://www.occitech.fr/
- http://roulezelectrique.com/
- http://antix.mepis.org/
- http://gagne.homedns.org/
- https://www.transip.eu/
- https://www.tilaa.com/
- http://www.quickvz.com/
- https://mapzen.com/
- http://services.gisgraphy.com/
- http://www.pagesjaunes.fr/
- http://www.yapaslefeuaulac.ch/
- http://geekfault.org/
- http://techtach.com/
- https://www.sqlite.org/
- https://sematext.com/blog/
- http://www.kris3d.com/
- https://stackoverflow.com/jobs
- http://www.timschwartz.org/
- https://tuxicoman.jesuislibre.net/
- http://blog.mro.name/
- http://mro.name/foaf.rdf

A title appears, but is truncated/not the right one:
- http://expandedramblings.com/index.php/3-reasons-self-hosted-video-better-youtube/
- http://www.amazon.fr/s/ref=nb_sb_noss?__mk_fr_FR=%C3%85M%C3%85%C5%BD%C3%95%C3%91&url=search-alias%3Daps&field-keywords=test
- http://www.meetup.com/
- https://wiki.cyanogenmod.org/w/Install_CM_for_jfltexx
- http://host1.no/

A title appears, but non-ASCII characters are replaced with a question mark (encoding issue?):
- http://www.upv.es/
- http://fr.lolix.org/search/offre/offre.php?id=21082

Furthermore, title fetching fails on links to PDF files.

### PR DESCRIPTION
**Pull request by ArthurHoaro**
see #531 for details

Almost everything is now working, except:

  * hosts which seem to have an anti-bot policy (2 in #531).
  * hosts which have a very high latency (1 in #531).

### CODE REVIEW

File: application/HttpUtils.php [hunk 1]
```diff
@@ -35,22 +35,33 @@ function get_http_response($url, $timeout = 30, $maxBytes = 4194304)
...
-                         .' Gecko/20100101 Firefox/23.0',
-            'request_fulluri' => true,
+            'user_agent' => 'Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:45.0)'
+                         .' Gecko/20100101 Firefox/45.0',
+            'accept_language' => setlocale(LC_COLLATE, 0) . ';en-US',
+        ),
+        'ssl' => array(
+            'verify_peer' => false,
+            'verify_peer_name' => false,
+            'allow_self_signed' => true
```
Comment (nodiscc) [inline application/HttpUtils.php]: I'm not sure at all we should ignore SSL verification entirely...

Comment (ArthurHoaro) [inline application/HttpUtils.php]: What's the potential risk?

**Comment by ArthurHoaro** [comment]
Update:

  * Better `Accept-Language` header.
  * Support of International Domain Names (requires `php-intl`).
  * Support xhtml charset definition:
```xhtml
<meta http-equiv="Content-Type" content="application/xhtml+xml;charset=iso-8859-1" />
```

File: tests/HttpUtils/GetHttpUrlTest.php [hunk 2]
```diff
@@ -35,4 +35,36 @@ public function testGetInvalidRemoteUrl()
...
+     */
+    public function testGetAbsoluteUrlWithAbsolute()
+    {
+        $origin = 'http://non.existent/blabla/?test';
+        $target = 'http://other.url/stuff.php';
+
+        $this->assertEquals($target, getAbsoluteUrl($origin, $target));
+    }
+
+    public function tmpTest()
```
Comment (virtualtam) [inline tests/HttpUtils/GetHttpUrlTest.php]: temporarily empty? :)

File: application/HttpUtils.php (hunk 1 above)
Comment (virtualtam) [inline application/HttpUtils.php]: interesting read: [PHP Security - Transport Layer Security](http://phpsecurity.readthedocs.org/en/latest/Transport-Layer-Security-%28HTTPS-SSL-and-TLS%29.html)

SSL should be enforced by default, with a possibility for users to disable it (at their own risk)

File: tests/HttpUtils/GetHttpUrlTest.php (hunk 2 above)
Comment (ArthurHoaro) [inline tests/HttpUtils/GetHttpUrlTest.php]: But I like my empty test methods. :(

File: application/HttpUtils.php (hunk 1 above)
Comment (nodiscc) [inline application/HttpUtils.php]: I can't find a serious risk off the the top of my head, but bypassing security mechanisms tends to not end well. I agree that this should be optional.

Comment (ArthurHoaro) [inline application/HttpUtils.php]: I did not take the time to read @virtualtam paper yet, but I don't think an option is necessary here. We can just remove TLS bypass.

File: application/HttpUtils.php [hunk 3]
```diff
@@ -60,16 +67,19 @@ function get_http_response($url, $timeout = 30, $maxBytes = 4194304)
 }
 
 /**
- * Retrieve HTTP headers, following n redirections (temporary and permanent).
+ * Retrieve HTTP headers, following n redirections (temporary and permanent ones).
  *
- * @param string $url initial URL to reach.
- * @param int $redirectionLimit max redirection follow..
+ * @param string $url              initial URL to reach.
+ * @param int    $redirectionLimit max redirection follow..
```
Comment (julienCXX) [inline application/HttpUtils.php]: 2 dots at the end?

File: application/Url.php [hunk 4]
```diff
@@ -221,6 +235,22 @@ public function cleanup()
     }
 
     /**
+     * Converts an URL with an International Domain Name host to a ASCII one.
+     * This requires PHP-intl. If it's not available, just returns this->cleanup().
+     *
+     * @return string converted cleaned up URL.
+     */
+    public function indToAscii()
```
Comment (julienCXX) [inline application/Url.php]: Shouldn’t it be called `idnToAscii` instead?

Comment (ArthurHoaro) [inline application/Url.php]: Thanks.

**Pull request merged**